app/models/custom_message_setting.rb

```diff
@@ -113,15 +113,17 @@ def self.nested_hash(hash=nil)
   def custom_message_keys_are_available
     return false if !value[:custom_messages].is_a?(Hash) || errors.present?
 
-    custom_messages_hash = {}
-    custom_messages.values.compact.each do |val|
-      custom_messages_hash = self.class.flatten_hash(custom_messages_hash.merge(val)) if val.is_a?(Hash)
+    en_translation_hash = self.class.flatten_hash(MessageCustomize::Locale.available_messages('en'))
+    custom_message_keys = custom_messages.values.sum([]){|val| self.class.flatten_hash(val).keys}.uniq
```

Comment:
`Array#sum` is supported by Ruby 2.4.0 or later. (https://ruby-doc.org/core-2.4.0/Array.html#method-i-sum)
Since Redmine 3.2 supports Ruby 1.9.3, this method should not be usable.

Reply:
Fix 0c7f436